controllers/argocd/deployment.go

```diff
@@ -801,9 +801,12 @@ func (r *ReconcileArgoCD) reconcileRepoDeployment(cr *argoprojv1a1.ArgoCD) error
 		deploy.Spec.Template.Spec.ServiceAccountName = cr.Spec.Repo.ServiceAccount
 	}
 
-	repoEnv := proxyEnvVars()
+	// Global proxy env vars go first
+	repoEnv := cr.Spec.Repo.Env
+	// Environment specified in the CR take precedence over everything else
+	repoEnv = argoutil.EnvMerge(repoEnv, proxyEnvVars(), false)
```

Comment:
We want to be able to overide the proxyEnvVars (cluster proxy env vars) with proxy env vars defined in the CR.  So, the merge func call should look like:
```
	repoEnv = argoutil.EnvMerge(proxyEnvVars(), repoEnv, true)
```

Reply:
IMHO, it does exactly that:

1. Set env to what is specified in CR
2. Try to merge results of `proxyEnvVars()` into `repoEnv`, but do not override  existing entries (`override` is `false`)

Reply:
OK.  You are right.  Both get the same result.